### ISSUE TITLE
Windows OpenGLES 2 does not work.

### ISSUE THREAD

**Issue by Ghoster738** [Ghoster738/Future-Cop-MIT#1]
If anyone has a solution for this issue, please tell me what I am doing wrong.

**Comment by Ghoster738** [comment]
I discovered that I could use Mesa LLVMpipe for the renderer. Yes, it is software, but it is better than no graphics.

The second and best option for performance is ANGLE which could translate OpenGLES 2 calls into Direct3D calls. Although, I had trouble with this, because I could not find instructions on how to compile it on Linux with MinGW-w64.

I could also use this for my other compiles, but Windows definitely needs a renderer.

This is the source [https://wiki.qt.io/Cross_compiling_Mesa_for_Windows](url)

**Comment by killermosi** [comment]
Is this issue still current?

**Comment by Ghoster738** [comment]
Yes, unfortunately.

I did not try compiling Mesa yet.

I attempted to get OpenGL working normally using MinGW-w64, GLEW, SDL2, and OpenGL 2 or OpenGLES 2. Non of them seems to be working so far. The executables with graphics seems to just crash for various reasons.

I tried to make the graphics initialization code work better. Before where just hacks to get them to work.

I am now going to try this cross compilation again.

**Comment by Ghoster738** [comment]
I got an idea, but my time is tight for this early week unfortunately.

Rather than trying to get my project to run with OpenGL, I will try to simply get an another SDL Windows compilation to be done using this YouTube video "[Setup] SDL Windows Setup with Mingw and Simple OpenGL Application | Introduction to SDL2" from Mike Shah. After I get this SDL2 complied and working, I will try to work on this project again using Glad. If this works I will use Glad rather than GLEW instead because it seems to come with the things I need due to it being configurable.

Again it will take me a few days to do, and this is mostly because my time is tight. I will start it when I have enough free time.

**Comment by Ghoster738** [comment]
> Is this issue still current?

This setup worked. I made a few mistakes. First, I naively though that MinGW would make projects that would always run on Wine. Well, I can confirm after debugging something about missing Windows functions I concluded that Wine version that I am using 5 does not support it. I do not know if they resolved this issue, but I do know that it works on Windows.

I simplified the compilation of my project for all platforms. GLEW is removed because it is no longer needed. Now, I am using Glad 2 now. I am happy that this build is finally almost done.

Now, I have a few problems preventing me from releasing the actual Windows version.
First is deciding which OpenGL32.dll to include. Preferably, I would like to include an OpenGL32.dll that I trust.
The other more pressing issue is that SDL2 is not giving Window borders to the two programs. Without these Window borders then it would be annoying for the user to turn off my programs.

I think there is something wrong with the SDL2 library that I am using. I wander if I am compiling them correctly.
